test/GameLogic/gameLogic.test.js

```diff
@@ -24,33 +49,58 @@ describe('GameLogic', () => {
             const positions = [1, 2, 3];
             positions.forEach((cell) => !positions[positions.length - 1] ? 
                 board.placeToken(playerToken, cell) : board.placeToken(new Token('O').getToken(), cell));    
-            const winDetected = game.detectRowWin();
+            const winDetected = game.containsWinningRow();
             expect(winDetected).toEqual(false);
         });
     });
 
-    describe('getWinningToken()', () => {
+    describe('containsWinningMainDiagonal()', () => {
         let board;
         let game;
-        const positions = [1, 2, 3];
-
+        const placeTokenFn = (cell) => board.placeToken(playerToken, cell)
+        
         beforeEach(() => {
             board = new Board();
             game = new GameLogic(board, playerToken);
         });
         
-        test('will return a string representation of the winning player\'s token', () => { 
-            const placeTokenFunc = (cell) => board.placeToken(playerToken, cell);
-            positions.forEach(placeTokenFunc);
-            const winner = game.getWinningToken();
-            expect(winner).toEqual('X');
+        test('will return a true boolean value when a main diagonal win is detected', () => {
+            const positions = [1, 5, 9]; // main diagonal positions (top-left  -> bottom-right) 
+            positions.forEach(placeTokenFn);
+            const winDetected = game.containsWinningMainDiagonal();
+            expect(winDetected).toBe(true);
         });
 
-        test('will return a null value if no winning player is found', () => {
-            positions.forEach((cell) => !positions[positions.length - 1] ? 
-                board.placeToken(playerToken, cell) : board.placeToken(new Token('O').getToken(), cell));
-            const winner = game.getWinningToken();
-            expect(winner).toEqual(null);
+        test('will return a false boolean value when a main diagonal win is not detected in the board', () => {
+            const positions = [1, 5]
+            positions.forEach(placeTokenFn); 
+            const winDetected = game.containsWinningCounterDiagonal();
```

Comment:
This should be `containsWinningMainDiagonal()`

Reply:
The method was changed and checked for accuracy. The code is in working state. :)